python/pyspark/sql/datasource.py

```diff
@@ -298,6 +305,90 @@ def read(self, partition: InputPartition) -> Iterator[Union[Tuple, Row]]:
         ...
 
 
+class DataSourceStreamReader(ABC):
```

Comment:
Should have versionadded directives, with some docstring

Reply:
Added.